diff --git a/sample/rtmp_server/Makefile b/sample/rtmp_server/Makefile
index f81e492..afe6fb8 100644
--- a/sample/rtmp_server/Makefile
+++ b/sample/rtmp_server/Makefile
@@ -16,7 +16,7 @@ CFLAGS += -I$(SDIR)/../test_mmf/maixcam_lib/include \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/media-server/libflv/include \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/media-server/librtmp/include \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/sdk/include
-CPPFLAGS += $(subst -std=gnu11,-std=gnu++11, $(CFLAGS)) $(INCS) \
+CPPFLAGS += $(subst rv64imafdcv0p7xthead,rv64imafdc, $(subst -std=gnu11,-std=gnu++11, $(CFLAGS))) $(INCS) \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/include \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/sdk/include \
 -I$(SDIR)/../test_mmf/media_server-1.0.x/sdk/libhttp/include \
@@ -110,7 +110,7 @@ $(SDIR)/%.o: $(SDIR)/%.c
 	@echo [$(notdir $(CC))] $(notdir $@)
 
 $(TARGET): mmflibs $(COMM_OBJ) $(OBJS) $(ISP_OBJ) $(MW_LIB)/libvenc.a $(MW_LIB)/libsys.a
-	@$(CXX) -o $@ -Wl,-rpath='$$ORIGIN/dl_lib' -Wl,-rpath=/mnt/system/usr/lib -Wl,-rpath=/mnt/system/usr/lib/3rd -Wl,--start-group -lflv -lrtmp $(OBJS) $(COMM_OBJS) -lsys $(MW_LIB)/libsys.a -Wl,--end-group -lmaixcam_lib $(filter-out -static, $(ELFFLAGS)) $(EXTRA_LDFLAGS)
+	@$(CXX) -o $@ -Wl,-rpath='$$ORIGIN/dl_lib' -Wl,-rpath=/mnt/system/usr/lib -Wl,-rpath=/mnt/system/usr/lib/3rd -Wl,--start-group -lflv -lrtmp $(OBJS) $(COMM_OBJS) -lsys $(MW_LIB)/libsys.a -Wl,--end-group -lmaixcam_lib $(subst rv64imafdcv0p7xthead,rv64imafdc, $(filter-out -static, $(ELFFLAGS))) $(EXTRA_LDFLAGS)
 	@echo -e $(BLUE)[LINK]$(END)[$(notdir $(CXX))] $(notdir $@)
 
 clean: clean_mmflibs
